src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs

```diff
@@ -103,7 +104,16 @@ private void EndFunction(FunctionInstanceLogEntry item)
             var function = startedEvent.FunctionMetadata;
             string eventName = success ? MetricEventNames.FunctionInvokeSucceeded : MetricEventNames.FunctionInvokeFailed;
             string functionName = function != null ? function.Name : string.Empty;
-            string data = string.Format(Microsoft.Azure.WebJobs.Script.Properties.Resources.FunctionInvocationMetricsData, startedEvent.FunctionMetadata.Language, functionName, success, Stopwatch.IsHighResolution);
+
+            // This has low cadinality but we allocate the string every call (even though ironically it's often not used due to rollups)
+            // It's cheaper to cache and lookup rather than generate and pay for GC here.
+            // Note: this would be faster with a readonly record struct key, but StyleCop is angry with it until an upcoming 1.2.x beta release
+            var key = (startedEvent.FunctionMetadata.Language, functionName, success, Stopwatch.IsHighResolution);
```

Comment:
Out of curiosity. From the grand scheme of things, does this really pay off in comparison to for example trying to use a faster way of allocating the string with for example `string.Create`? Now this requires a not so small ValueTuple and a dictionary lookup which also comes with runtime costs.

Reply:
Good question! Yes it pays off, because that string we're allocating below is _far bigger_, and it needs to be garbage collected later - to boot, it's often not even used.

This is one of those things were "if you do X a lot, even in gen0, it isn't free" - that's still cost to cleanup the mess later. I'd like to get load tests as part of PRs in general, so that amortized cost of GC is measured and recognized.